File: client/src/main/java/au/com/codeka/warworlds/client/opengl/Sprite.java
package au.com.codeka.warworlds.client.opengl;

import com.google.common.base.Preconditions;

/** A {@link Sprite} is basically a quad + texture. */
public class Sprite extends SceneObject {
  private final SpriteTemplate tmpl;
  private float alpha;

  public Sprite(DimensionResolver dimensionResolver, SpriteTemplate tmpl) {
    super(dimensionResolver);
    this.alpha = 1.0f;
    this.tmpl = Preconditions.checkNotNull(tmpl);
  }

  public void setAlpha(float alpha) {
    this.alpha = alpha;
  }

  public float getAlpha() {
    return alpha;
  }

  @Override
  protected void drawImpl(float[] mvpMatrix) {
    tmpl.draw(mvpMatrix, alpha);
  }
}

File: client/src/main/kotlin/au/com/codeka/warworlds/client/opengl/Sprite.kt
package au.com.codeka.warworlds.client.opengl

/** A [Sprite] is basically a quad + texture.  */
class Sprite(
    dimensionResolver: DimensionResolver, debugName: String, private val tmpl: SpriteTemplate) :
    SceneObject(dimensionResolver, "Sprite:$debugName") {
  var alpha = 1.0f

  override fun drawImpl(mvpMatrix: FloatArray?) {
    tmpl.draw(mvpMatrix, alpha)
  }
}
